core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -452,16 +452,21 @@ public void removeBlocks(List<Long> blockIds, boolean delete) throws Unavailable
   @Override
   public void validateBlocks(Function<Long, Boolean> validator, boolean repair)
       throws UnavailableException {
-    List<Long> invalidBlocks = new ArrayList<>();
-    for (long blockId : mBlocks.keySet()) {
-      if (!validator.apply(blockId)) {
-        invalidBlocks.add(blockId);
+    List<Long> invalidBlocks =
+        mBlocks.keySet().stream().filter((blockId) -> !validator.apply(blockId))
+            .collect(Collectors.toList());
+    if (!invalidBlocks.isEmpty()) {
+      List<Long> loggedBlocks = invalidBlocks.stream().limit(100).collect(Collectors.toList());
+      LOG.warn("Found {} blocks without corresponding file metadata. Blocks include {}.",
```

Comment:
I think we want to be explicit about this not being the full list

Reply:
fixed PTAL